crates/aggregator/src/lib.rs

```diff
@@ -23,36 +23,55 @@ use eigen_services_blsaggregation::bls_agg::{
     AggregateReceiver, BlsAggregatorService, ServiceHandle,
 };
 use eigen_services_operatorsinfo::operatorsinfo_inmemory::OperatorInfoServiceInMemory;
+use eigen_task_processor::task::Task;
+use eigen_task_processor::task_manager::TaskManagerContract;
+use eigen_task_processor::IndexingTaskProcessor;
 use futures_util::{future, StreamExt};
 use rpc_server::{ProcessSignedTaskResponse, ProcessSignedTaskResponseServer};
-use std::{net::SocketAddr, sync::Arc};
+use std::fmt::Debug;
+use std::net::SocketAddr;
 use tarpc::server::{self, Channel};
 use tarpc::tokio_serde::formats::Json;
-use tokio::sync::Mutex;
 use tracing::info;
 
+use alloy::{
+    contract::private::{Provider as ProviderTrait, Transport},
+    network::Network,
+};
 pub use config::AggregatorConfig;
 pub use eigen_services_blsaggregation::{
     bls_agg::TaskMetadata, bls_aggregation_service_response::BlsAggregationServiceResponse,
 };
 pub use error::AggregatorError;
 pub use signed_task_response::SignedTaskResponse;
-pub use traits::{
-    task_processor::{TaskProcessor, TaskProcessorError},
-    task_response::TaskResponse,
-};
 
 /// Aggregator
 #[derive(Debug)]
-pub struct Aggregator<TP> {
+pub struct Aggregator<TM, T, P, N>
+where
+    TM: TaskManagerContract<T, P, N> + Debug + Send + Sync + 'static + Clone,
+    T: Transport + Clone + Send + Sync + 'static,
+    P: ProviderTrait<T, N> + Clone + 'static,
+    N: Network,
+{
     port_address: String,
-    task_processor: Arc<Mutex<TP>>,
+    task_processor: IndexingTaskProcessor<TM, T, P, N>,
     service_handle: ServiceHandle,
     aggregated_response_receiver: AggregateReceiver,
     ws_rpc_url: String,
 }
 
-impl<TP: TaskProcessor + Send + Sync + 'static + Clone> Aggregator<TP> {
+impl<TM, T, P, N> Aggregator<TM, T, P, N>
```

Comment:
Can we keep the `TaskProcessor` trait and make it so `IndexedTaskProcessor` implements that trait instead?